File: Java/Mahasiswa.java
public class Mahasiswa extends Sivitas_Akademik{
    // Nama Mahasiswa
    // private String nama;
    // Nim Mahasiswa
    private String nim;
    // Nim Mahasiswa
    // private String Jenis_kelamin;
    // Program Studi Mahasiswa
    private String prodi;
    // Fakultas Mahasiswa
    private String fakultas;

    Mahasiswa(){} // konstruktor Mahasiswa

    Mahasiswa(String nim, String prodi, String fakultas) {
        // konstruktor langsung isi attribut
        this.nim = nim;
        this.prodi = prodi;
        this.fakultas = fakultas;
    }

    // get set
    // public void setNama(String namaMahasiswa) {
    //     this.nama = namaMahasiswa;
    // }

    // public String getNama() {
    //     return nama;
    // }

    public void setNim(String nimMahasiswa) {
        this.nim = nimMahasiswa;
    }

    public String getNim() {
        return nim;
    }

    // public void setJenis_kelamin(String Jenis_kelamin) {
    //     this.Jenis_kelamin = Jenis_kelamin;
    // }

    // public String getJenis_kelamin() {
    //     return this.Jenis_kelamin;
    // }

    public void setProdi(String programStudi) {
        this.prodi = programStudi;
    }

    public String getProdi() {
        return prodi;
    }
    
    public void setFakultas(String fakultas) {
        this.fakultas = fakultas;
    }

    public String getFakultas() {
        return this.fakultas;
    }
}

File: C++/Mahasiswa.cpp
// Import library.
#include <bits/stdc++.h>
#include "Sivitas_Akademik.cpp"
// Using standard namespace.
using namespace std;

// membuat kelas Mahasiswa
class Mahasiswa : public Sivitas_Akademik
{
private:
    // Membuat Private attributes.
    string nim;
    string prodi;
    string fakultas;

public:
    Mahasiswa() {} // konstruktor Mahasiswa
    Mahasiswa(string nama, string nik, string Jenis_kelamin, string asal_universitas, string email_edu, string nim, string prodi, string fakultas)
        : Sivitas_Akademik(nik, nama, Jenis_kelamin, asal_universitas, email_edu) // memanggil konstruktor dari kelas induk (superclass)
    {

        // konstruktor langsung isi attribut
        this->nim = nim;
        this->prodi = prodi;
        this->fakultas = fakultas;
    }

    // Setter and Getter methods for private attributes

    // Set Nim
    void setNim(string nimMahasiswa)
    {
        this->nim = nimMahasiswa;
    }

    // Get Nim
    string getNim()
    {
        return nim;
    }

    // Set Prodi
    void setProdi(string programStudi)
    {
        this->prodi = programStudi;
    }

    // Get Prodi
    string getProdi()
    {
        return prodi;
    }

    // Set Fakultas
    void setFakultas(string fakultas)
    {
        this->fakultas = fakultas;
    }

    // Get Fakultas
    string getFakultas()
    {
        return this->fakultas;
    }

    // Destruktor
    ~Mahasiswa()
    {
    }
};